soundbay/inference.py

```diff
@@ -134,8 +134,45 @@ def inference_to_file(
     output_file = output_path / filename
     concat_dataset.to_csv(index=False, path_or_buf=output_file)
 
+    # save raven file
+    threshold = 0.5
+    thresholdtext = int(threshold*10)
+    raven_filename = f"raven_annotations-{datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}-{model_name}-{dataset_name}-thresh0{thresholdtext}.csv"
+    raven_output_file = output_path / raven_filename
+    raven_df = inference_csv_to_raven(results_df, test_dataset, threshold=threshold)
+    raven_df.to_csv(index=False, path_or_buf=raven_output_file, sep="\t")
+
     return
 
+def inference_csv_to_raven(probsdataframe, test_dataset, threshold=0.5):
+    # transforms the probability dataframe to a raven format with class_1 predictions as the annotated bounding boxes
+    len_dataset = probsdataframe.shape[0] #number of segments in wav
+    seq_len = test_dataset.seq_length
+    all_begin_times = np.arange(0, len_dataset * seq_len, seq_len)
+    positives = []
+    for index, row in probsdataframe.iterrows():
+        if row['class1_prob'] > threshold:
+            positives.append(index)
+
+    begin_times = np.round(all_begin_times[positives],decimals=1)
```

Comment:
btw, why we need round or decimals=1? what's gonna happen we if want seq_length of 0.125?

Reply:
When I didn't add the rounding I got something messed up (e.g. 24.0000001 sec or something). Probably has something to do with the float variable

Reply:
When will we need such precise seq lengths? Is there significant data in the realm of 0.01 seconds?

Reply:
I tried it again and it seems that the non rounded numbers don't affect the visual result in raven. I'm not sure if it might affect evaluation calculations later, but if it will then we'll change it back.